lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -1059,6 +1059,8 @@ def hwaddr(value, query='', alias='hwaddr'):
 
     try:
         v = netaddr.EUI(value)
+    except netaddr.AddrFormatError:
+        raise errors.AnsibleFilterError(alias + ': not a hardware address: %s' % value)
```

Comment:
How about returning False instead of raising error. As per [documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html#mac-address-filter) if string is not a valid mac address then we should return False

Reply:
Okay.

```python
PLAY [hwaddr validation] *****************************************************************************************************************************

TASK [assert] ****************************************************************************************************************************************
fatal: [localhost]: FAILED! => {
    "assertion": "vars['maddr'] | hwaddr()",
    "changed": false,
    "evaluated_to": false,
    "msg": "'maddr' is not maddr"
}

PLAY RECAP *******************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1   
```

Reply:
@Akasurde?